mergesort.js

```diff
@@ -1,49 +1,6 @@
-function mergesort(a){
-  
-  if (a.length == 1){
-    return a;
-  }
-  else if (a.length == 2){
-    if (a[0] > a[1]){
-      return [a[1], a[0]];
-    }
-    else{
-      return a;
-    }
-  }
-
-  var p = parseInt(a.length/2);
-  var m1 = mergesort(a.slice(0, p));
-  var m2 = mergesort(a.slice(p));
-
-  var ret = [];
-
-  while(1){
-    if(m1.length > 0 && m2.length > 0){
-      if(m1[0] <= m2[0]){
-        ret.push(m1[0]);
-        m1 = m1.slice(1);
-      }
-      else{
-        ret.push(m2[0]);
-        m2 = m2.slice(1);
-      }
-    }
-    else if(m1.length > 0){
-      ret = ret.concat(m1);
-      m1 = [];
-    }
-    else if (m2.length > 0){
-      ret = ret.concat(m2);
-      m2 = [];
-    }
-    else{
-      break;
-    }
-  }
-
-  return ret;
+var mergesort = a => a.length < 2 ? a: merge(mergesort(a.splice(0, a.length/2)),mergesort(a))
```

Comment:
hmm, the second merge sort is being called on a. does a change?

Reply:
Hey, yes... that's the trick! Splice changes the original array, so we don't have to create two new arrays and waste precious resources :)

Reply:
Check this out => JS gets into the first place! 🥇

Reply:
still more chars than the J version (243 v 172) and the J's not golfed or squished into one line hah

Reply:
Yeah, J is the shorter one in terms of chars 👍  But JS is way more readable, anyone with enough programming experience can easily figure out what's going on